src/modules/stakewise/stakewise-daemon/server/status/get-validator-statuses.go

```diff
@@ -62,28 +63,33 @@ func (c *statusGetValidatorsStatusesContext) PrepareData(data *swapi.ValidatorSt
 	if err != nil {
 		return fmt.Errorf("error getting public keys: %w", err)
 	}
-	statuses, err := bc.GetValidatorStatuses(context.Background(), publicKeys, nil)
+	statusResponse, err := bc.GetValidatorStatuses(context.Background(), publicKeys, nil)
 	if err != nil {
 		return fmt.Errorf("error getting validator statuses: %w", err)
 	}
 
+	registeredPubkeys := make([]beacon.ValidatorPubkey, 0)
+	for _, pubkeyStatus := range nodesetStatusResponse {
+		registeredPubkeys = append(registeredPubkeys, pubkeyStatus.Pubkey)
+	}
+
 	beaconStatuses := make(map[string]types.ValidatorState)
 	nodesetStatuses := make(map[string]swapi.NodesetStatus)
 
 	for _, pubKey := range publicKeys {
-		status, exists := statuses[pubKey]
+		status, exists := statusResponse[pubKey]
 		if exists {
 			beaconStatuses[pubKey.HexWithPrefix()] = status.Status
 		}
 	}
 
 	for _, pubKey := range publicKeys {
 		switch {
-		case IsRegisteredToStakewise(pubKey, statuses):
+		case IsRegisteredToStakewise(pubKey, statusResponse):
```

Comment:
These should probably all be `nodesetStatusResponse` now since I'm assuming that's what you're going to use for them.